ui/src/app/components/app-router.tsx

```diff
@@ -16,8 +17,24 @@ export const usePath = () => {
   return path;
 };
 
-export const withRouteData = WrappedComponent => ({routeData, ...props}) => {
-  routeDataStore.set(routeData);
+// TODO angular2react: This isn't really the right place to be making the store updates but it's the
+// best place I found while we're using both angular and react routers
+export const withRouteData = WrappedComponent => ({intermediaryRoute = false, routeData, ...props}) => {
+  const params = useParams();
+
+  useEffect(() => {
+    if (!intermediaryRoute) {
+      routeConfigDataStore.next(routeData);
+      routeDataStore.set(routeData);
+    }
+  }, [routeData]);
```

Comment:
nit: technically these should both also depend on intermediaryRoute - though understood that's static